### PR TITLE
remove deprecation warnings from SHA.jl

### PR DESCRIPTION
**Pull request by jakebolewski**
Fix deprecation warnings

### CODE REVIEW

File: src/base_functions.jl [hunk 1]
```diff
@@ -19,18 +19,18 @@ Ch(x,y,z)  = (((x) & (y)) $ ((~(x)) & (z)))
...
+sigma0_512(x) =   (S64( 1, @compat(UInt64(x))) $ S64( 8, @compat(UInt64(x))) $ R( 7,   @compat(UInt64(x))))
+sigma1_512(x) =   (S64(19, @compat(UInt64(x))) $ S64(61, @compat(UInt64(x))) $ R( 6,   @compat(UInt64(x))))
 
 # Let's be able to bswap arrays of these types as well
-bswap!(x::Vector{Uint32})  = map!(bswap, x)
-bswap!(x::Vector{Uint64})  = map!(bswap, x)
-bswap!(x::Vector{Uint128}) = map!(bswap, x)
+bswap!(x::Vector{UInt32})  = map!(bswap, x)
+bswap!(x::Vector{UInt64})  = map!(bswap, x)
+bswap!(x::Vector{UInt128}) = map!(bswap, x)
```
Comment (staticfloat) [inline src/base_functions.jl]: Do we not need to `@compat(UInt32)` here?

Comment (staticfloat) [inline src/base_functions.jl]: Oh, right, because it's the type and not the constructor.  I'm jetlagged.  :P

**Comment by staticfloat** [comment]
LGTM!  Thanks!

**Comment by jakebolewski** [comment]
@staticfloat can you tag a new version?

**Comment by staticfloat** [comment]
Done.
-E

**Comment by jakebolewski** [comment]
Thanks!

**Pull request merged**